File: admin-portal/src/types/product.ts
export interface Product {
  id: number
  name: string
  description?: string
  price: number // Note: For production, consider using a decimal library (e.g., decimal.js) or string representation
  currency: string // ISO 4217 currency code (e.g., "USD", "EUR", "GBP")
  stockQuantity?: number
  imageUrl?: string
  category?: string
  createdAt: string // ISO 8601 format date string (e.g., "2024-01-14T10:30:00Z")
  updatedAt: string // ISO 8601 format date string (e.g., "2024-01-14T10:30:00Z")
}

export interface ProductRequest {
  name: string
  description?: string
  price: number // Note: For production, consider using a decimal library (e.g., decimal.js) or string representation
  currency?: string // ISO 4217 currency code (defaults to "USD" if not provided)
  stockQuantity?: number
  imageUrl?: string
  category?: string
}

File: backend/src/main/java/com/shophub/model/Product.java
package com.shophub.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(nullable = false, length = 3)
    private String currency = "USD"; // ISO 4217 currency code (e.g., USD, EUR, GBP)

    private Integer stockQuantity;
    private String imageUrl;
    private String category;

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
